service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -362,4 +364,15 @@ public ApplicationConfigProperties applicationConfigProperties() {
     public HealthCheckServer healthCheckServer() {
         return new VertxBasedHealthCheckServer(vertx(), applicationConfigProperties());
     }
+
+    /**
+     * Creates a new instance of {@link ResourceLimitChecks} based on prometheus metrics data.
+     * 
+     * @return A ResourceLimitChecks instance.
+     */
+    @Bean
+    @ConfigurationProperties(prefix = "hono.plan.prometheusBased")
+    public ResourceLimitChecks resourceLimitChecks() {
+        return new PrometheusBasedResourceLimitChecks(vertx());
```

Comment:
I guess this means that it is no longer possible to run Hono without Prometheus because we always inject the Prometheus based checks which will always fail if Prometheus is not running, right?

Reply:
@sophokles73 How about a similar approach that we use for the metrics? During maven build we provide an profile `metrics-prometheus` so that prometheus will be used to store the metrics. And then corresponding beans are created based on condition `@ConditionalOnClass(name = "io.micrometer.prometheus.PrometheusMeterRegistry")`.

In this case it looks like this. If no prometheus is being used, then this bean won't be created.
```
    @Bean
    @ConditionalOnClass(name = "io.micrometer.prometheus.PrometheusMeterRegistry")
    @ConfigurationProperties(prefix = "hono.plan.prometheusBased")
    public ResourceLimitChecks resourceLimitChecks() {
        return new PrometheusBasedResourceLimitChecks(vertx());
    }
```

Reply:
the problem with this approach is that it would still require Prometheus running. The Maven profile is (so far) only used to add the Prometheus specific Meter registry to the class path. That alone makes no assumption whether Prometheus will actually be running. this is because Prometheus actively *scrapes* the metrics from components instead of having the components report metrics to Prometheus.
FMPOV it would be better to use a `ConditionalOnProperty` and check for the Prometheus host name being set. The default yaml should then (of course) not contain the corresponding properties.

Reply:
@sophokles73 `ConditionalOnProperty` sounds good.